src/tool_operate.c

```diff
@@ -1393,10 +1393,11 @@ static CURLcode operate_do(struct GlobalConfig *global,
           my_setopt(curl, CURLOPT_SSL_ENABLE_ALPN, 0L);
         }
 
-        /* new in 7.40.0 */
+        /* new in 7.40.0, abstract support added in 7.53.0 */
         if(config->unix_socket_path)
-          my_setopt_str(curl, CURLOPT_UNIX_SOCKET_PATH,
-                        config->unix_socket_path);
+          my_setopt_str(curl, config->abstract_unix_socket ?
+                        CURLOPT_ABSTRACT_UNIX_SOCKET :
+                        CURLOPT_UNIX_SOCKET_PATH, config->unix_socket_path);
```

Comment:
Also it occurs to me that CURLoption parameter is stringized by the function-like macro so I'm pretty sure this is not going to work, run with --libcurl to confirm. Assuming that's the case do this instead
~~~c
if(config->unix_socket_path) {
  if(config->abstract_unix_socket) {
    my_setopt_str(curl, CURLOPT_ABSTRACT_UNIX_SOCKET,
                  config->unix_socket_path);
  }
  else {
    my_setopt_str(curl, CURLOPT_UNIX_SOCKET_PATH,
                  config->unix_socket_path);
  }
}
~~~